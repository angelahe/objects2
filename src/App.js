import React, { Component } from 'react';
import './App.css';
//import Starter from './components/Starter';
import MathComp from './components/MathComp';
import AccountUI from './components/AccountUI';
import AccountsUI from './components/AccountsUI';
import CommunityUI from './components/CommunityUI';
import LinkedListUI from './components/LinkedListUI';

import NavBar from './components/NavBar';

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            navClicked: "default",
            message: "Starting message"
        };
    }

    onNavClickChanged = (clickBtn) => {
//        console.log("in onNavClickChanged, clickbtn is", clickBtn);
        this.setState({navClicked : clickBtn, message : clickBtn + " was clicked"});
    }

    render() {

        return (

            <div className="App">
                <NavBar navClicked={this.onNavClickChanged}/>
                {(this.state.navClicked==="linked") ? <LinkedListUI /> : null}
                {(this.state.navClicked==="math") ? <MathComp /> : null}
                {(this.state.navClicked==="account") ? <AccountUI /> : null}
                {(this.state.navClicked==="accounts") ? <AccountsUI /> : null}
                {(this.state.navClicked==="community") ? <CommunityUI/> : null}
            </div>
        );
    }
}

export default App;
