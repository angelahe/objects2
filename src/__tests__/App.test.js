import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import {shallow} from "enzyme/build";
import renderer from "react-test-renderer";

const props = new App();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<App />);
});

test('App renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Starter App renders correctly', () => {
  props.state.NavClicked = "default";
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Math App renders correctly', () => {
  props.state.NavClicked = "math";
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Account App renders correctly', () => {
  props.state.NavClicked = "account";
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Accounts App renders correctly', () => {
  props.state.NavClicked = "accounts";
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Communities App renders correctly', () => {
  props.state.NavClicked = "community";
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});