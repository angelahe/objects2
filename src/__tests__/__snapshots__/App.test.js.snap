// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account App renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <header>
      <img
        alt="one"
        className="App-icon"
        navbtn="default"
        onClick={[Function]}
        src="analytics.svg"
      />
      <img
        alt="two"
        className="App-icon"
        navbtn="math"
        onClick={[Function]}
        src="creative.svg"
      />
      <img
        alt="three"
        className="App-icon"
        navbtn="account"
        onClick={[Function]}
        src="leader.svg"
      />
      <img
        alt="four"
        className="App-icon"
        navbtn="accounts"
        onClick={[Function]}
        src="teamwork.svg"
      />
      <img
        alt="five"
        className="App-icon"
        navbtn="community"
        onClick={[Function]}
        src="teamwork2.svg"
      />
    </header>
    <p />
  </div>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;

exports[`Accounts App renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <header>
      <img
        alt="one"
        className="App-icon"
        navbtn="default"
        onClick={[Function]}
        src="analytics.svg"
      />
      <img
        alt="two"
        className="App-icon"
        navbtn="math"
        onClick={[Function]}
        src="creative.svg"
      />
      <img
        alt="three"
        className="App-icon"
        navbtn="account"
        onClick={[Function]}
        src="leader.svg"
      />
      <img
        alt="four"
        className="App-icon"
        navbtn="accounts"
        onClick={[Function]}
        src="teamwork.svg"
      />
      <img
        alt="five"
        className="App-icon"
        navbtn="community"
        onClick={[Function]}
        src="teamwork2.svg"
      />
    </header>
    <p />
  </div>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <header>
      <img
        alt="one"
        className="App-icon"
        navbtn="default"
        onClick={[Function]}
        src="analytics.svg"
      />
      <img
        alt="two"
        className="App-icon"
        navbtn="math"
        onClick={[Function]}
        src="creative.svg"
      />
      <img
        alt="three"
        className="App-icon"
        navbtn="account"
        onClick={[Function]}
        src="leader.svg"
      />
      <img
        alt="four"
        className="App-icon"
        navbtn="accounts"
        onClick={[Function]}
        src="teamwork.svg"
      />
      <img
        alt="five"
        className="App-icon"
        navbtn="community"
        onClick={[Function]}
        src="teamwork2.svg"
      />
    </header>
    <p />
  </div>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;

exports[`Communities App renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <header>
      <img
        alt="one"
        className="App-icon"
        navbtn="default"
        onClick={[Function]}
        src="analytics.svg"
      />
      <img
        alt="two"
        className="App-icon"
        navbtn="math"
        onClick={[Function]}
        src="creative.svg"
      />
      <img
        alt="three"
        className="App-icon"
        navbtn="account"
        onClick={[Function]}
        src="leader.svg"
      />
      <img
        alt="four"
        className="App-icon"
        navbtn="accounts"
        onClick={[Function]}
        src="teamwork.svg"
      />
      <img
        alt="five"
        className="App-icon"
        navbtn="community"
        onClick={[Function]}
        src="teamwork2.svg"
      />
    </header>
    <p />
  </div>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;

exports[`Math App renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <header>
      <img
        alt="one"
        className="App-icon"
        navbtn="default"
        onClick={[Function]}
        src="analytics.svg"
      />
      <img
        alt="two"
        className="App-icon"
        navbtn="math"
        onClick={[Function]}
        src="creative.svg"
      />
      <img
        alt="three"
        className="App-icon"
        navbtn="account"
        onClick={[Function]}
        src="leader.svg"
      />
      <img
        alt="four"
        className="App-icon"
        navbtn="accounts"
        onClick={[Function]}
        src="teamwork.svg"
      />
      <img
        alt="five"
        className="App-icon"
        navbtn="community"
        onClick={[Function]}
        src="teamwork2.svg"
      />
    </header>
    <p />
  </div>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;

exports[`Starter App renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <header>
      <img
        alt="one"
        className="App-icon"
        navbtn="default"
        onClick={[Function]}
        src="analytics.svg"
      />
      <img
        alt="two"
        className="App-icon"
        navbtn="math"
        onClick={[Function]}
        src="creative.svg"
      />
      <img
        alt="three"
        className="App-icon"
        navbtn="account"
        onClick={[Function]}
        src="leader.svg"
      />
      <img
        alt="four"
        className="App-icon"
        navbtn="accounts"
        onClick={[Function]}
        src="teamwork.svg"
      />
      <img
        alt="five"
        className="App-icon"
        navbtn="community"
        onClick={[Function]}
        src="teamwork2.svg"
      />
    </header>
    <p />
  </div>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
